cmd/clusterctl/cmd/root.go

```diff
@@ -20,45 +20,98 @@ import (
 	"flag"
 	"fmt"
 	"os"
+	"strings"
 
+	"github.com/MakeNowJust/heredoc"
 	"github.com/spf13/cobra"
-	cliflag "k8s.io/component-base/cli/flag"
 	"k8s.io/klog"
 )
 
-const deprecationMsg string = "NOTICE: clusterctl has been deprecated in v1alpha2 and will be removed in a future version."
-
-var helpTemplate = fmt.Sprintf(`%s
-{{with (or .Long .Short)}}{{. | trimTrailingWhitespaces}}
-
-{{end}}{{if or .Runnable .HasSubCommands}}{{.UsageString}}{{end}}`, deprecationMsg)
+var cfgFile string
 
 var RootCmd = &cobra.Command{
 	Use:   "clusterctl",
-	Short: "cluster management",
-	Long:  `Simple kubernetes cluster management`,
-	RunE: func(cmd *cobra.Command, args []string) error {
-		return cmd.Help()
-	},
+	Short: "clusterctl controls a management cluster for Cluster API",
+	Long: LongDesc(`
+		Get started with Cluster API using clusterctl for initializing a management cluster by installing
+		cluster API providers, and then use clusterctl for creating yaml templates for your workload clusters.`),
 }
 
 func Execute() {
 	if err := RootCmd.Execute(); err != nil {
+		//TODO: print error stack if log v>0
+		//TODO: print cmd help if validation error
 		fmt.Fprintln(os.Stderr, err)
 		os.Exit(1)
 	}
 }
 
-func exitWithHelp(cmd *cobra.Command, err string) {
-	_, _ = fmt.Fprintln(os.Stderr, err)
-	_ = cmd.Help()
-	os.Exit(1)
-}
-
 func init() {
+
 	klog.InitFlags(flag.CommandLine)
-	RootCmd.SetGlobalNormalizationFunc(cliflag.WordSepNormalizeFunc)
 	RootCmd.PersistentFlags().AddGoFlagSet(flag.CommandLine)
-	RootCmd.SetHelpTemplate(helpTemplate)
-	InitLogs()
+
+	// hiding all the klog flags except
+	// --log_dir
+	// --log_file
+	// --log_file_max_size
+	// -v, --v Level
+
+	RootCmd.PersistentFlags().MarkHidden("alsologtostderr")  //nolint
+	RootCmd.PersistentFlags().MarkHidden("log_backtrace_at") //nolint
+	RootCmd.PersistentFlags().MarkHidden("logtostderr")      //nolint
+	RootCmd.PersistentFlags().MarkHidden("stderrthreshold")  //nolint
+	RootCmd.PersistentFlags().MarkHidden("vmodule")          //nolint
+	RootCmd.PersistentFlags().MarkHidden("skip_log_headers") //nolint
+	RootCmd.PersistentFlags().MarkHidden("skip_headers")     //nolint
+	RootCmd.PersistentFlags().MarkHidden("add_dir_header")   //nolint
+
+	// makes logs look nicer for a CLI app
+	RootCmd.PersistentFlags().Set("skip_headers", "true") //nolint
+	RootCmd.PersistentFlags().Set("logtostderr", "true")  //nolint
+
+	RootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "Path to the the clusterctl config file (default is $HOME/.clusterctl.yaml)")
+}
+
+const Indentation = `  `
+
+// LongDesc normalizes a command's long description to follow the conventions.
+func LongDesc(s string) string {
```

Comment:
I would encourage you to use `dedendt.Dedent` here. Kubernetes already has a precedent for this library for this exact situation:

https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/config.go#L235